addons/purchase_stock/models/stock.py

```diff
@@ -10,6 +10,8 @@ class StockPicking(models.Model):
 
     purchase_id = fields.Many2one('purchase.order', related='move_lines.purchase_line_id.order_id',
         string="Purchase Orders", readonly=True)
+    # Technical field to activate cache mechanism on aggregating computed fields to recompute
+    purchase_ids = fields.Many2many('purchase.order', string="Inversed field for purchase.order")
```

Comment:
maybe add store=False

Reply:
The table already exists, having `store=False` doesn't change anything, no?

https://github.com/odoo/odoo/blob/966410eb08b0c8c278dffd673f0042b7b0b427d3/addons/purchase_stock/models/purchase.py#L22

Reply:
you are right, but you should add the same relation
in stock.py:
`purchase_ids = fields.Many2many('purchase.order', relation='purchase_order_stock_picking_rel' column1='stock_picking_id', column2='purchase_order_id')`

in purchase.py
`picking_ids = fields.Many2many('stock.picking', relation='purchase_order_stock_picking_rel' column1='purchase_order_id', column2='stock_picking_id', ...)`

Note : the table `purchase_order_stock_picking_rel` is already created.

Reply:
ORM already does this automatically. You only need to do it manually if you have few many2many relation between two tables.

https://github.com/odoo/odoo/blob/00025dd406b8b7b6322672000e0d6401d769e73c/odoo/fields.py#L3652-L3660

Specifying the names explicitly looks attractive, but it's actually a source of potential typos and broken deployments